isotovideo

```diff
@@ -89,13 +134,30 @@ sub _get_version_string () {
     return "Current version is $thisversion [interface v$OpenQA::Isotovideo::Interface::version]";
 }
 
+sub _exit_code_from_test_results () {
+    my @results = glob(path(bmwqemu::result_dir(), "result-*.json"));
+    return EXIT_STATUS_ERR_NO_TESTS if @results == 0;
+
+    my $did_fail = 0;
+    for my $result_file_path (@results) {
+        my $result_file = path($result_file_path);
+        my $test_result = decode_json($result_file->slurp)->{result};
+        diag sprintf("Test result [%s] %s", $result_file->to_string, $test_result);
+        # If a failure (anything different from ok & softfail) is found, keep it.
+        next if $did_fail;
+
+        $did_fail = !!($test_result !~ m/^(ok|softfail)$/);
```

Comment:
This looks pretty like you just want to exit the loop if you get one failure, so why not:
```suggestion
        $did_fail = !!($test_result !~ m/^(ok|softfail)$/) and last;
```

Reply:
Hmm, now I'm torn

On one side I'd like to get the message for all test results.

But on the other side the result is determined by the first failure 🤔

Reply:
Oh I see. I missed that. But you can remove the `!!`

Reply:
Applied to [267cb9d6](https://github.com/os-autoinst/os-autoinst/commit/267cb9d6ba265a92f26a32f760ed587b866251f2)